lib/Locale/Country/Extra.pm

```diff
@@ -33,7 +34,14 @@ sub code_from_country {
     my ($self, $country) = @_;
 
     my %code_countries = reverse %{ $self->_country_codes };
-    return lc $code_countries{$country};
+
+    unless( exists $code_countries{$country} ){
+        $country = lc $country;
+        $country = first { $country eq lc $_ } keys %code_countries;
```

Comment:
I'd suggest expanding to `List::Util::first` here if this is the only instance where we use this function so we limit importing; if so, we can also reduce the `use` declaration above to `use List::Util ();`.